compiler/rustc_parse/src/parser/path.rs

```diff
@@ -412,13 +420,34 @@ impl<'a> Parser<'a> {
     }
 
     /// Parses a single argument in the angle arguments `<...>` of a path segment.
-    fn parse_angle_arg(&mut self) -> PResult<'a, Option<AngleBracketedArg>> {
+    fn parse_angle_arg(&mut self, _ident_sp: Span) -> PResult<'a, Option<AngleBracketedArg>> {
         let lo = self.token.span;
         let arg = self.parse_generic_arg()?;
         match arg {
             Some(arg) => {
                 if self.check(&token::Colon) | self.check(&token::Eq) {
-                    let (ident, gen_args) = self.get_ident_from_generic_arg(arg, lo)?;
+                    let (ident, gen_args) = match self.get_ident_from_generic_arg(arg, lo) {
+                        Ok(ident_gen_args) => ident_gen_args,
+                        Err(arg) => return Ok(Some(AngleBracketedArg::Arg(arg))),
+                        /*
+                        Err(mut err) => {
+                            if self.check(&token::Eq) {
+                                // We could have incorrectly parsed an equality constraint
+                                // when there is actually a missing closing angle bracket in an
+                                // assignment such as in `Vec<(u32, _, _) = vec![]`
+                                let span = ident_sp.to(self.prev_token.span);
+                                let source = self.sess.source_map().span_to_snippet(span).unwrap();
+                                err.span_suggestion(
+                                    span,
+                                    "possibly missing closing angle bracket",
+                                    format!("try to use the following: {}>", source),
+                                    Applicability::MaybeIncorrect,
+                                );
```

Comment:
You would likely want to use this here in this way to get the effect you're looking for:
```suggestion
                                err.span_suggestion(
                                    span.shrink_to_hi(),
                                    "possibly missing closing angle bracket",
                                    ">".to_string(),
                                    Applicability::MaybeIncorrect,
                                );
```

Reply:
Thanks for the suggestion.